lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_to_s3.rb

```diff
@@ -143,10 +138,12 @@ def self.available_options
             end,
             optional: true,
             type: Symbol,
-            default_value: nil, # Using nil under the hood until we remove skip_if_exists
+            # We cannot set a default value and have backward compatibility with skip_if_exists at the same time.
+            # (Short of duplicating the default value knowledge in the action implementation)
+            #
+            # We have a test for the default behavior which should hopefully remind us to uncomment this line once well remove skip_if_exists.
+            # default_value: :fail,
             verify_block: proc do |value|
-              next if value.nil?
-
```

Comment:
Given that the `default_value` of this `if_exists` option end up being `nil` (you removed the explicit `default_value` on line 146, but that `ConfigItem` being `optional: true`, the default default value (😅) will still be `nil`)

As such, I think it's very possible that someone calling the action without an explicit value for `if_exists` will end up having this `verify_block` called with `nil` as a `value`? (except if fastlane and `ConfigItem` already handle the case of "value not passed explicitly" in a special way that doesn't call `verify_block` in those cases? 🤔  Which would explain why the Unit Tests still pass?)

The fact that the CI is green and tests from line 148 still pass (instead of raising from within `verify_block` due to the value `nil` not being one of the 3 allowed values) still makes me pause, I think it's worth checking what's happening here, to confirm if that's a special behavior of fastlane not calling `verify_block` in those cases vs our Unit Test or our condition being incorrect and we're missing something.

Reply:
Seems like Fastlane already accounts for the scenario of `optional: true` and `verify_block`.

I added an explicit test for it in [b261eef](https://github.com/wordpress-mobile/release-toolkit/pull/500/commits/b261eef85ecb909d2c4cb85ba1197ed87ff93536) and also checked by running the action with only the required params. If failed because I didn't have the right AWS credentials set, which to me is confirmation that the `verify_block` was bypassed.

<details>
<summary>Stacktrace</summary>

```
➜ bf run upload_to_s3 file:./README.md bucket:test

[✔] 🚀
[20:57:32]: --------------------------
[20:57:32]: --- Step: upload_to_s3 ---
[20:57:32]: --------------------------

Looking for related GitHub issues on fastlane/fastlane...

➡️  Fastlane supply aab to internal track fail with google api error bad request
    https://github.com/fastlane/fastlane/issues/18670 [closed] 31 💬
    20 Oct 2021

➡️  [Supply] Upload to Beta track failed when a version in Internal Test track is having Release state
    https://github.com/fastlane/fastlane/issues/18303 [closed] 2 💬
    03 Jul 2021

➡️  [Supply] Google Api Error: badRequest: User fraction must be lower than one
    https://github.com/fastlane/fastlane/issues/17605 [closed] 5 💬
    19 Jan 2021

and 2 more at: https://github.com/fastlane/fastlane/search?q=Aws%3A%3AS3%3A%3AErrors%3A%3ABadRequest&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
bundler: failed to load command: fastlane (/Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/bin/fastlane)
Traceback (most recent call last):
	48: from /Users/gio/.rbenv/versions/2.7.4/bin/bundle:23:in `<main>'
	47: from /Users/gio/.rbenv/versions/2.7.4/bin/bundle:23:in `load'
	46: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/exe/bundle:36:in `<top (required)>'
	45: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/lib/bundler/friendly_errors.rb:120:in `with_friendly_errors'
	44: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/exe/bundle:48:in `block in <top (required)>'
	43: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/lib/bundler/cli.rb:25:in `start'
	42: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	41: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/lib/bundler/cli.rb:31:in `dispatch'
	40: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	39: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	38: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	37: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/lib/bundler/cli.rb:486:in `exec'
	36: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/lib/bundler/cli/exec.rb:23:in `run'
	35: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/lib/bundler/cli/exec.rb:58:in `kernel_load'
	34: from /Users/gio/.rbenv/versions/2.7.4/lib/ruby/gems/2.7.0/gems/bundler-2.3.24/lib/bundler/cli/exec.rb:58:in `load'
	33: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/bin/fastlane:23:in `<top (required)>'
	32: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/bin/fastlane:23:in `load'
	31: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/bin/fastlane:23:in `<top (required)>'
	30: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/cli_tools_distributor.rb:123:in `take_off'
	29: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/commands_generator.rb:43:in `start'
	28: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/commands_generator.rb:354:in `run'
	27: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	26: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
	25: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	24: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	23: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	22: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/commands_generator.rb:226:in `block (2 levels) in run'
	21: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/one_off.rb:22:in `execute'
	20: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/one_off.rb:42:in `run'
	19: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	18: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	17: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	16: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	15: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	14: from /Users/gio/Developer/a8c/release-toolkit/lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_to_s3.rb:23:in `run'
	13: from /Users/gio/Developer/a8c/release-toolkit/lib/fastlane/plugin/wpmreleasetoolkit/actions/common/upload_to_s3.rb:60:in `file_is_already_uploaded?'
	12: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-s3-1.121.0/lib/aws-sdk-s3/client.rb:7249:in `head_object'
	11: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-core-3.171.0/lib/seahorse/client/request.rb:72:in `send_request'
	10: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-core-3.171.0/lib/seahorse/client/plugins/response_target.rb:24:in `call'
	 9: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-core-3.171.0/lib/aws-sdk-core/plugins/response_paging.rb:12:in `call'
	 8: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-core-3.171.0/lib/seahorse/client/plugins/request_callback.rb:71:in `call'
	 7: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-core-3.171.0/lib/aws-sdk-core/plugins/param_converter.rb:26:in `call'
	 6: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-core-3.171.0/lib/aws-sdk-core/plugins/idempotency_token.rb:19:in `call'
	 5: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-core-3.171.0/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:16:in `call'
	 4: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-core-3.171.0/lib/aws-sdk-core/plugins/checksum_algorithm.rb:111:in `call'
	 3: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-s3-1.121.0/lib/aws-sdk-s3/plugins/accelerate.rb:43:in `call'
	 2: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-s3-1.121.0/lib/aws-sdk-s3/plugins/dualstack.rb:21:in `call'
	 1: from /Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-s3-1.121.0/lib/aws-sdk-s3/plugins/sse_cpk.rb:24:in `call'
/Users/gio/Developer/a8c/release-toolkit/vendor/bundle/ruby/2.7.0/gems/aws-sdk-core-3.171.0/lib/seahorse/client/plugins/raise_response_errors.rb:17:in `call': Aws::S3::Errors::BadRequest (Aws::S3::Errors::BadRequest)
```

</details>